### DISCUSSION TITLE
Coding

### DISCUSSION THREAD

**Discussion by blank-repo** [UniversalMediaServer/UniversalMediaServer#4141] [Q&A] [answered]
Hey so, I've been modifying UMS to create a personal feature similar to Live Subtitles and I've hit a snag.
I don't know how to refresh the hierachy after adding resources to the real folder, I've looked all over source but I'm officially lost :/

```
New file somefile.mkv was detected and added to the Media Library
INFO  03:28:34.461 [File event] The file X:\somepath\somefolder\somefile.mkv could not be parsed. It will be hidden
INFO  03:28:34.549 [File event] The file X:\somepath\somefolder\somefile.mkv could not be parsed. It will be hidden
```

New files are added... but only visible if I navigate to a secondary *unopened* folder also housing them.
How do I update the folder (or virtual folder) I used/opened to trigger adding them?

Thank you in advance for any insight you may be able to provide :]

**Comment by SubJunk** [comment]
@RocketFriday welcome to UMS! Cool idea. I'll try to help
How are you accessing the folder? Browser or DLNA renderer?

**Reply by blank-repo** [reply]
DLNA renderer.

**Reply by blank-repo** [reply]
If you want, msging [me on discord](https://discord.com/users/148253593076498432) sends a notification to my phone, or we could hop in a call and talk about it.

**Reply by blank-repo** [reply, marked as answer]
WOOOOOW, are you kidding me! xD
So I looked closer at the debug log and then at the live subs source and I couldn't really tell what I was doing different.
Then I saw that I was doing: 
`originalResource.getParent().addChild(item);`
afterall, I had logged that out to confirm it was the folder I wanted and it was.
but I'm guessing by simply doing:
`addChild(item);`
It's signalling the renderer properly... or maybe I had the wrong folder... idk. But it works now! 
Thanks for replying and making me feel like help is available! :]
